test/userController.test.js

```diff
@@ -93,3 +127,240 @@ describe('GET /api/v1/users/:user_id endpoint', () => {
+
+    // Assert: Check the response status for success
+    expect(response.status).toBe(200);
+    expect(response.body).toHaveProperty('msg', 'Success! Password Updated.');
+  });
+
+  it('should return an error if old password is incorrect', async () => {
+    // Arrange: Log in and get a signed cookie with valid test user credentials
+    const signedCookie = await loginAndReturnCookie(testUserCredentials);
+
+    // Arrange: Define the incorrect old password and a new password
+    const incorrectOldPassword = 'incorrectSecret';
+    const newPassword = 'newSecret';
+
+    // Act: Attempt to update the user's password with an incorrect old password
+    const response = await request(app)
+      .patch('/api/v1/users/updateUserPassword')
+      .set('Cookie', [signedCookie])
+      .send({ oldPassword: incorrectOldPassword, newPassword });
+
+    // Assert: Check the response status for an error
+    expect(response.status).toBe(401);
+  });
+
+  it('should return an error if required data is missing', async () => {
+    // Arrange: Log in and get a signed cookie with valid test user credentials
+    const signedCookie = await loginAndReturnCookie(testUserCredentials);
+
+    // Act: Attempt to update the user's password without providing old and new passwords
+    const response = await request(app)
+      .patch('/api/v1/users/updateUserPassword')
+      .set('Cookie', [signedCookie])
+      .send({});
+
+    // Assert: Check the response status for an error
+    expect(response.status).toBe(400);
+  });
+});
+
+describe('DELETE /api/v1/users/deleteSingleUser endpoint', () => {
+  it('should delete a user if the user exists', async () => {
+    // Arrange: Log in and get a signed cookie with valid test user credentials
+    const signedCookieAdmin = await loginAndReturnCookie(testAdminCredentials);
+
+    const response = await request(app)
+      .delete(`/api/v1/users/deleteSingleUser/${testUser.id}`) // Assuming testUser is the user you want to delete
+      .set('Cookie', [signedCookieAdmin]);
+
+    // Assert: Check the response status for success
+    expect(response.status).toBe(200);
+    expect(response.body).toHaveProperty('msg', 'Success! User Deleted.');
+  });
+
+  it('should return an error if the user does not exist', async () => {
+    // Arrange: Log in and get a signed cookie with valid test user credentials
+    const signedCookieAdmin = await loginAndReturnCookie(testAdminCredentials);
+
+    // Act: Attempt to delete a non-existent user
+    const response = await request(app)
+      .delete('/api/v1/users/nonexistentUserId')
+      .set('Cookie', [signedCookieAdmin]);
+
+    // Assert: Check the response status for an error
+    expect(response.status).toBe(404);
+  });
+
+  it('should return an error if not authenticated', async () => {
+    // Act: Attempt to delete a user without authentication
+    const response = await request(app).delete(
+      '/api/v1/users/deleteSingleUser'
+    );
+
```

Comment:
Ah I think we're mixing up the controller handler name `deleteSingleUser` with the actual URL/path that is defined in `userRoutes.js`. We can also verify by double checking what the correct path is in the generated API docs.

Reply:
Yes, I wasn't very attentive